python-sdk/src/astro/files/locations/base.py

```diff
@@ -109,12 +119,24 @@ def check_non_existing_local_file_path(path: str) -> bool:
         return True
 
     @staticmethod
-    def get_location_type(path: str) -> FileLocation:
+    def get_location_type(path: str, conn_id: str | None = None) -> FileLocation:
         """Identify where a file is located
 
         :param path: Path to a file in the filesystem/Object stores
+        :param conn_id: Airflow connection id
         """
+        connection_type = ""
+        if conn_id is not None:
+            connection_type = BaseHook.get_connection(conn_id).conn_type
+
         file_scheme = urlparse(path).scheme
+
+        file_scheme_from_conn_id = CUSTOM_CONN_TYPE_TO_FILE_SCHEME.get(connection_type)
```

Comment:
@sunank200 I think we should unit test case for this method. WDYT?

Reply:
added: https://github.com/astronomer/astro-sdk/pull/1478/commits/5a0ed3ff0b89713f7d39867754857b31fd01e423